hudi-client/hudi-client-common/src/test/java/org/apache/hudi/config/TestHoodieWriteConfig.java

```diff
@@ -725,6 +726,30 @@ void testBloomFilterType() {
     assertEquals(bloomFilterType, config.getBloomFilterType());
   }
 
+  @Test
+  public void testStreamingWritesToMetadataConfig() {
+    Properties props = new Properties();
+    HoodieWriteConfig config = HoodieWriteConfig.newBuilder()
+        .withPath("/tmp")
+        .withProperties(props)
+        .withEngineType(EngineType.SPARK).build();
+
+    assertTrue(config.isStreamingWritesToMetadataEnabled(HoodieTableVersion.EIGHT));
+    assertFalse(config.isStreamingWritesToMetadataEnabled(HoodieTableVersion.SIX));
+
+    config = HoodieWriteConfig.newBuilder()
+        .withPath("/tmp")
+        .withProperties(props)
+        .withEngineType(EngineType.FLINK).build();
+    assertFalse(config.isStreamingWritesToMetadataEnabled(HoodieTableVersion.SIX));
```

Comment:
Add an assertion for table version 8 as well here and below

Reply:
sure